### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by fabiosantoscode**

### CODE REVIEW

File: index.css [hunk 1]
```diff
@@ -0,0 +1,48 @@
+@import "@economist/component-typography";
+@import "@economist/component-palette";
+@import "@economist/component-grid";
+@import "@economist/component-icon";
+
+.bar-wrapper a, .bar-wrapper a:visited {
```
Comment (umbobabo) [inline index.css]: I personally prefer a new line after the comma.

File: index.css [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  color: inherit;
+  text-decoration: none;
+  border-bottom: 1px solid var(--color-london);
```
Comment (umbobabo) [inline index.css]: Can we use a modifier to add the border? Maybe is more flexible for all the possible context.

File: index.css [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  width: 2em;
+}
+
+.bar-wrapper--close-wrapper {
+  position: absolute;
+  right: var(--barwrapper-close-button-right-spacing, var(--grid-gutter-s));
+
+  cursor: pointer;
+
+  /* Compensate Icon's white space */
```
Comment (umbobabo) [inline index.css]: Can you test the component with component-icon >= 5.0.0.
Maybe this part will be not anymore necessary.

File: index.css [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+   * The icon is supposed to have 1em of height, but it currently has
+   * 1em of useless space surrounding it. */
+  height: 2em;
+  width: 2em;
+}
+
+.bar-wrapper--close-wrapper {
+  position: absolute;
+  right: var(--barwrapper-close-button-right-spacing, var(--grid-gutter-s));
+
```
Comment (umbobabo) [inline index.css]: No need for blank lines.

File: index.es6 [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+    let classNames = [ 'bar-wrapper' ];
+    let closedClassNames = [ 'bar-wrapper--closed' ];
+    let containerClassNames = [ 'bar-wrapper--container' ];
+    let closeWrapperClassNames = [ 'bar-wrapper--close-wrapper' ]
+    let closeClassNames = [ 'bar-wrapper--close' ];
+
+    if (className) {
+      classNames = classNames.concat([ className ]);
+      closedClassNames = closedClassNames.concat([ `${className}--closed` ]);
```
Comment (umbobabo) [inline index.es6]: I will not use className here as prefix. Classname can be something like "classA classB classC", not just one string.
I think is probably best have a "prefix" props to use for this purpose.

File: index.es6 [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+    if (this.state && this.state.closed) {
+      return (<div className={closedClassNames.join(' ')}></div>);
+    }
+
+    let closeButton = null;
+
+    if (close) {
+      closeButton = (
+        <span onClick={this.onClose}
```
Comment (umbobabo) [inline index.es6]: Ideally what is inside here should be replaced by a component-link-button that can automatically provide the icon in a more compact way, but I'm not sure that tabIndex will be properly propagated to the A tag.

File: index.css (hunk 3 above)
Comment (fabiosantoscode) [inline index.css]: Yes! Let me check it out!

File: index.es6 (hunk 5 above)
Comment (fabiosantoscode) [inline index.es6]: Good point!

File: index.es6 (hunk 6 above)
Comment (fabiosantoscode) [inline index.es6]: component-link-button would be pretty nice, but we're not really doing anything that requires a href. Besides, the browser considers anything with a tabindex focusable

Comment (fabiosantoscode) [inline index.es6]: Anything :crown:

**Pull request merged**